### ISSUE TITLE
suggestion : use macro when constant expression

### ISSUE THREAD

**Issue by wighawag** [markknol/hx-verbal-expressions#2]
Hi, this looks like a cool idea, 
Since in many situation regular expression use constant strings a macro could generate the regular expression at compile time.
I guess it is already possible to do on top of your library but just thinking it could be a good adition to it too.

**Comment by markknol** [comment]
Yes I have been thinking of this too, it would be awesome. But I'm not sure how to do this exactly yet. Ideas?

**Comment by wighawag** [comment]
Did not have much a thought, and it might not be trivial. My first thought would be to chain a serie of macro static function with the same name (startOfline, then...) which will return a regexp expression

``` haxe
var regExpr = VerbalMacro.regexp().startOfLine()
   .then()
  .maybe("s")
  .then("://")
  .maybe("www.")
  .anythingBut(" ")
  .endOfLine();
```

then with the use  static extension the generated regexp could be augmented with the function you use 

``` haxe
regExpr.isMatch(url)
```

**Comment by markknol** [comment]
@simn suggested this: https://gist.github.com/Simn/01748a98af07128ee166 
Ill look into this soon.

**Comment by wighawag** [comment]
Looks great.
